### PR TITLE
IDENTITY-6611: Add more unit test cases for scim2 common utils

### PR DESCRIPTION
**Pull request by ashensw**

### CODE REVIEW

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/test/utils/CommonTestUtils.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    public static void initPrivilegedCarbonContext(String tenantDomain) throws Exception {
+        int tenantID = -1234;
+        String userName = "testUser";
+
+        initPrivilegedCarbonContext(tenantDomain, tenantID, userName);
+    }
+
+    public static void initPrivilegedCarbonContext() throws Exception {
+        String tenantDomain = "carbon.super";
```
Comment (harsha1979) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/test/utils/CommonTestUtils.java]: Use Carbon MultitenantConstants for these.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/test/utils/CommonTestUtils.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        int tenantID = -1234;
```
Comment (harsha1979) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/test/utils/CommonTestUtils.java]: Use Carbon MultitenantConstants for these.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/test/utils/CommonTestUtils.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+public class CommonTestUtils {
+
+    private CommonTestUtils() {
+    }
+
+    public static void initPrivilegedCarbonContext(String tenantDomain, int tenantID, String userName) throws Exception {
+
+        String carbonHome = Paths.get(System.getProperty("user.dir"), "target").toString();
+        System.setProperty("carbon.home", carbonHome);
```
Comment (harsha1979) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/test/utils/CommonTestUtils.java]: Use CarbonBaseConstants for these.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtilsTest.java [hunk 4]
```diff
@@ -129,4 +140,81 @@ public void testGetPrimaryFreeGroupName(String groupName, String expectedResult)
...
+    @Test
+    public void testGetGlobalConsumerId() throws Exception {
+        String tenantDomain = "testTenantDomain";
+
+        try {
+            initPrivilegedCarbonContext(tenantDomain);
+
+            Assert.assertEquals(SCIMCommonUtils.getGlobalConsumerId(), tenantDomain);
+
+        } finally {
```
Comment (harsha1979) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtilsTest.java]: So many line spaces..

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/test/utils/CommonTestUtils.java (hunk 1 above)
Comment (ashensw) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/test/utils/CommonTestUtils.java]: Updated the PR.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/test/utils/CommonTestUtils.java (hunk 2 above)
Comment (ashensw) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/test/utils/CommonTestUtils.java]: Updated the PR.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/test/utils/CommonTestUtils.java (hunk 3 above)
Comment (ashensw) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/test/utils/CommonTestUtils.java]: Updated the PR.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/test/utils/CommonTestUtils.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+    public static void initPrivilegedCarbonContext(String tenantDomain, int tenantID, String userName) throws Exception {
+
+        String carbonHome = Paths.get(System.getProperty("user.dir"), "target").toString();
+        System.setProperty(CarbonBaseConstants.CARBON_HOME, carbonHome);
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain);
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantID);
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setUsername(userName);
+
```
Comment (pulasthi7) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/test/utils/CommonTestUtils.java]: this new line not needed

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtilsTest.java [hunk 6]
```diff
@@ -129,4 +139,74 @@ public void testGetPrimaryFreeGroupName(String groupName, String expectedResult)
...
+                {true, true},
+                {false, false}
+        };
+    }
+
+    @Test
+    public void testUnsetThreadLocalToSkipSetUserClaimsListeners() throws Exception {
+
+        SCIMCommonUtils.unsetThreadLocalToSkipSetUserClaimsListeners();
+        Assert.assertNull(SCIMCommonUtils.getThreadLocalToSkipSetUserClaimsListeners());
```
Comment (pulasthi7) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtilsTest.java]: static import Assert.*

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/test/utils/CommonTestUtils.java (hunk 5 above)
Comment (ashensw) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/test/utils/CommonTestUtils.java]: Updated the PR.

**Pull request merged**